src/libs/domain/matchingContext.ts

```diff
@@ -57,18 +49,28 @@ export const doesTabContentMatchExpression = async (
       ])
     : Promise.resolve(true);
 
-export const filterContextsMatchingTabContent = async (
-  tab: Tab,
-  matchingContexts: MatchingContext[]
-) => {
+export const filterContexts = async (
+  matchingContexts: MatchingContext[],
+  tab: Tab
+): Promise<MatchingContext[]> => {
   const responses = await Promise.all(
-    matchingContexts.map(({ xpath, id }: MatchingContext) => {
-      return (
-        typeof xpath === 'undefined' ||
-        doesTabContentMatchExpression(tab, xpath, id)
-      );
+    matchingContexts.map((matchingContext: MatchingContext):
+      | Promise<MatchingContext | false>
```

Comment:
Could the return of this map be simplified to something simpler like `Promise<MatchingContext | null>` ?

Reply:
Sorry I did not see your comment @lutangar

Done !